apidef/api_definitions.go

```diff
@@ -543,25 +544,48 @@ func (a *APIDefinition) DecodeFromDB() {
 	}
 }
 
-func (s *StringRegexMap) Check(value string) string {
+func (s *StringRegexMap) Check(value string) (match string) {
+	if s.matchRegex == nil {
+		return
+	}
+
 	return s.matchRegex.FindString(value)
 }
 
-func (s *StringRegexMap) FindStringSubmatch(value string) []string {
-	return s.matchRegex.FindStringSubmatch(value)
+func (s *StringRegexMap) FindStringSubmatch(value string) (matched bool, match []string) {
+	if s.matchRegex == nil {
```

Comment:
I think this clause, and in `Check` function not needed anymore.

Reply:
If `s.matchRegex, err = regexp.Compile(s.MatchPattern);` fails, it will be `nil` so doesn't it panic ?